File: payroll-api/src/main/java/com/company/payroll/model/SystemAccount.java
package com.company.payroll.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
public class SystemAccount {
    private Integer aId;
    private String username;
    private String password;
    private String secretKey;
    private String roles;
    private LocalDateTime dateCreated;
    private LocalDateTime dateModified;
    private LocalDateTime lastLogin;
    private Byte lastAttempt;
    private Byte accountStatus;
    private String imgPath;
    private Integer staffId;
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/model/SystemAccount.kt
package com.company.payroll.model

import com.company.payroll.annotation.NoArg
import java.time.LocalDateTime

@NoArg
data class SystemAccount(
    var aId: Int,
    var username: String,
    var password: String?,
    var secretKey: String?,
    var roles: String?,
    var dateCreated: LocalDateTime?,
    var dateModified: LocalDateTime?,
    var lastLogin: LocalDateTime?,
    var lastAttempt: UByte?,
    var accountStatus: UByte?,
    var imgPath: String?,
    var staffId: Int?
)
